board.h

```diff
@@ -1,13 +1,24 @@
+#include <bitset> //for debugging
 #include <cstdlib>
 #include <iostream>
 #include <string>
+#include <unordered_set>
 #include <vector>
 
 struct Position {
 	unsigned int xpos = 0;
 	unsigned int ypos = 0;
 
-	bool operator==(const Position &b) const;
+	bool operator==(const Position &pos) const;
+};
+
+struct PositionHash { // hashing implementation for Position

```

Comment:
#nice loved it, awesome!

<img width="600" height="830" alt="image" src="https://github.com/user-attachments/assets/94b7d3ce-e232-444d-8e3f-b040cf9fb6c6" />